sdk/storage/CMakeLists.txt

```diff
@@ -65,13 +80,15 @@ add_library(azure-storage ${AZURE_STORAGE_HEADER} ${AZURE_STORAGE_SOURCE})
 
 find_package(LibXml2 REQUIRED)
 
-target_include_directories(azure-storage PUBLIC ${LIBXML2_INCLUDE_DIR} $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/inc> $<INSTALL_INTERFACE:include/azure_storage>)
+target_include_directories(azure-storage PUBLIC ${LIBXML2_INCLUDE_DIR} $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/inc> $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/external> $<INSTALL_INTERFACE:include/azure_storage>)
 
 target_link_libraries(azure-storage azure-core ${LIBXML2_LIBRARIES})
 
 if(MSVC)
     target_link_libraries(azure-storage bcrypt)
     target_compile_definitions(azure-storage PRIVATE NOMINMAX)
+    # C28020 and C28204 are introduced by nlohmann/json
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd28204 /wd28020")
```

Comment:
Will `target_compile_options` work?

Reply:
Yes, can you elaborate on why target_compile_options  is prefered in CMAKE_CXX_FLAGS?
I'll change to use target_compile_options in next iteration.

Reply:
The same reason as why we don't want to add these flags to all the targets in this whole repository.

by `target_compile_options` we only apply this flag to this single target. But by `CMAKE_CXX_FLAGS`, we apply this flag to all targets in this CMake and every CMake files by `add_subdirctory`.

Reply:
```
#pragma warning(push)
#pragma warning(disable : 28020 26439 6011)
#include "json.hpp"
#pragma warning(pop)
```

This works for me. Can you give it a try?

Reply:
You can also use `add_compile_options` (https://cmake.org/cmake/help/latest/command/add_compile_options.html)

Reply:
@katmsft can you please at least brief why this won't work or why you won't take it before resolving this conversation?

Reply:
First, this is already changed to `target_compile_options `, that's why I resolved it the first time. We have talked offline about why using `#pragma warning(disable : 28204)` won't work offline so I resolved it second time, sorry for not posting anything here.
The reason is that the warning is thrown in `vcruntime_exception.h`, which makes it impossible to guard as it needs to be guarded the first time this system library is included.